TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonContentAuthSigner.java

```diff
@@ -12,35 +12,49 @@
 public class NettyJsonContentAuthSigner extends ChannelInboundHandlerAdapter {
     IAuthTransformer.StreamingFullMessageTransformer signer;
     HttpJsonMessageWithFaultingPayload httpMessage;
-    List<HttpContent> receivedHttpContents;
+    List<HttpContent> httpContentsBuffer;
 
     public NettyJsonContentAuthSigner(IAuthTransformer.StreamingFullMessageTransformer signer) {
         this.signer = signer;
-        this.receivedHttpContents = new ArrayList<>();
+        this.httpContentsBuffer = new ArrayList<>();
     }
 
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
         if (msg instanceof HttpJsonMessageWithFaultingPayload) {
             httpMessage = (HttpJsonMessageWithFaultingPayload) msg;
         } else if (msg instanceof HttpContent) {
-            receivedHttpContents.add(((HttpContent) msg).retainedDuplicate());
```

Comment:
how can you guarantee that you can shed this retain?  My rule is that if a class is holding some refCnted resource for an asynchronous activity, it should do a retain.  It looks like the class didn't adequately release in error cases, but I'm not sure that shedding the retain is completely safe.

I guess the argument is that if we're not propagating it, then we're not releasing it, so as long as we hold back on firing it to the next handler, we're fine.  Please confirm.

Reply:
>if we're not propagating it, then we're not releasing it (yet)

Yes, this seemed to best align to me with the implicit contracts with handlers.

Reply:
I'm not counting this as an asynchronous activity since it happens in context with the embedded channel processing.

I didn't add this, but it may be helpful to include a `msg.touch()` here